File: VB/ReadOnlyFormFields/Form2.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.XtraScheduler


Namespace ReadOnlyFormFields
	Partial Public Class Form2
		Inherits DevExpress.XtraScheduler.UI.AppointmentForm
		Public Sub New(ByVal control As SchedulerControl, ByVal apt As Appointment, ByVal openRecurrenceForm As Boolean)
			MyBase.New(control, apt, openRecurrenceForm)
			Me.tbSubject.Enabled = False
			Me.edtEndDate.Enabled = False
			Me.edtEndTime.Enabled = False
		End Sub
		Protected Overrides Sub chkAllDay_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs)
			MyBase.chkAllDay_EditValueChanged(sender, e)
			Me.edtEndDate.Enabled = False
			Me.edtEndTime.Enabled = False
		End Sub
	End Class

End Namespace

File: CS/ReadOnlyFormFields/Form2.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.XtraScheduler;


namespace ReadOnlyFormFields {
    public partial class Form2 : DevExpress.XtraScheduler.UI.AppointmentForm {
        public Form2(SchedulerControl control, Appointment apt, bool openRecurrenceForm)
            : base(control, apt, openRecurrenceForm) {
            this.tbSubject.Enabled = false;
            this.edtEndDate.Enabled = false;
            this.edtEndTime.Enabled = false;
        }
        protected override void chkAllDay_EditValueChanged(object sender, EventArgs e) {
            base.chkAllDay_EditValueChanged(sender, e);
            this.edtEndDate.Enabled = false;
            this.edtEndTime.Enabled = false;
        }
    }
    
}
